go.mod

```diff
@@ -0,0 +1,40 @@
+module sigs.k8s.io/cluster-api-provider-aws
+
+go 1.12
+
+require (
+	cloud.google.com/go v0.36.0 // indirect
+	github.com/aws/aws-sdk-go v1.19.18
+	github.com/awslabs/goformation v0.0.0-20180916202949-d42502ef32a8
+	github.com/go-logr/logr v0.1.0
+	github.com/gogo/protobuf v1.2.1 // indirect
+	github.com/golang/groupcache v0.0.0-20190129154638-5b532d6fd5ef // indirect
+	github.com/golang/mock v1.2.0
+	github.com/gophercloud/gophercloud v0.0.0-20190225152240-2c53651e4c14 // indirect
+	github.com/gregjones/httpcache v0.0.0-20190212212710-3befbb6ad0cc // indirect
+	github.com/onsi/ginkgo v1.8.0
+	github.com/onsi/gomega v1.5.0
+	github.com/pkg/errors v0.8.1
+	github.com/prometheus/client_model v0.0.0-20190129233127-fd36f4220a90 // indirect
+	github.com/prometheus/common v0.2.0 // indirect
+	github.com/prometheus/procfs v0.0.0-20190225181712-6ed1f7e10411 // indirect
+	github.com/sanathkr/yaml v1.0.0 // indirect
+	github.com/spf13/cobra v0.0.3
+	github.com/spf13/pflag v1.0.3
+	golang.org/x/net v0.0.0-20190613194153-d28f0bde5980
+	golang.org/x/oauth2 v0.0.0-20190220154721-9b3c75971fc9 // indirect
+	k8s.io/api v0.0.0-20190222213804-5cb15d344471
+	k8s.io/apimachinery v0.0.0-20190703205208-4cfb76a8bf76
+	k8s.io/client-go v10.0.0+incompatible
+	k8s.io/cluster-bootstrap v0.0.0-20190223141759-fab9a0a63c55
+	k8s.io/code-generator v0.0.0-20181117043124-c2090bec4d9b
+	k8s.io/klog v0.3.2
+	k8s.io/kubernetes v1.13.3
+	sigs.k8s.io/cluster-api v0.1.8
+	sigs.k8s.io/controller-runtime v0.1.12
+	sigs.k8s.io/controller-tools v0.1.11
+	sigs.k8s.io/testing_frameworks v0.1.1
+	sigs.k8s.io/yaml v1.1.0
+)
+
+replace k8s.io/apimachinery => k8s.io/apimachinery v0.0.0-20190221213512-86fb29eff628
```

Comment:
Do we need to list `cluster-api` as a replace to avoid issues after we start cutting v1alpha2-related releases?

Reply:
I'd think that given the incompatibilities it'll not default to v2. We can add a replace at some point if needed though

Reply:
I'm not sure how much compatibility testing is done with gomod, especially prior to major versions being set. It definitely will not attempt to compile against versions like `dep` does.

Reply:
We can follow-up in another commit if needed, I opened this mostly to backport the modules support before the release is cut.